implementations/rust/ockam/ockam_command/src/node/show.rs

```diff
@@ -218,44 +223,72 @@ pub async fn is_node_up(
     node_client: &mut BackgroundNodeClient,
     wait_until_ready: bool,
 ) -> Result<bool> {
-    let attempts = match wait_until_ready {
-        true => IS_NODE_UP_MAX_ATTEMPTS,
-        false => 1,
-    };
+    let node_name = node_client.node_name();
+    if !is_node_accessible(ctx, node_client, wait_until_ready).await? {
+        warn!(%node_name, "the node is not accessible in time");
+        return Ok(false);
+    }
+    if !is_node_ready(ctx, node_client, wait_until_ready).await? {
+        warn!(%node_name, "the node is not ready in time");
+        return Ok(false);
+    }
+    Ok(true)
+}
 
-    let retries =
-        FixedInterval::from_millis(IS_NODE_UP_TIME_BETWEEN_CHECKS_MS as u64).take(attempts);
+/// Return true if the node is accessible via TCP
+async fn is_node_accessible(
+    ctx: &Context,
+    node_client: &mut BackgroundNodeClient,
+    wait_until_ready: bool,
+) -> Result<bool> {
+    let retries = FibonacciBackoff::from_millis(IS_NODE_ACCESSIBLE_TIME_BETWEEN_CHECKS_MS);
 
-    let now = std::time::Instant::now();
     let node_name = node_client.node_name();
 
+    let mut total_time = Duration::from_secs(0);
     for timeout_duration in retries {
-        // The node is down if its default tcp listener has not been started yet
-        let node = node_client.cli_state().get_node(&node_name).await.ok();
-        let node_tcp_listener_address = node.as_ref().and_then(|n| n.tcp_listener_address());
-
-        if node.is_none() || node_tcp_listener_address.is_none() {
-            trace!(%node_name, "node has not been initialized");
-            tokio::time::sleep(timeout_duration).await;
-            continue;
+        if total_time >= IS_NODE_ACCESSIBLE_TIMEOUT || !wait_until_ready && !total_time.is_zero() {
+            return Ok(false);
+        };
+        if node_client.is_accessible(ctx).await.is_ok() {
+            trace!(%node_name, "node is accessible");
+            return Ok(true);
         }
+        trace!(%node_name, "node is not accessible");
```

Comment:
Instead of doing the `!wait_until_ready && !total_time.is_zero()` above, wouldn't it be the same to check `if !wait_until_ready { return Ok(false) }` right after this line? And the same for the `is_node_ready` function.

Reply:
We still need to make sure that we run the loop at least once, which is what `!total_time.is_zero()`  is testing.